File: java/455. Assign Cookies.java
import java.util.Arrays;

class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int noContentChildren = 0;
        int sIndex = 0;
        while(noContentChildren < g.length && sIndex < s.length){
            int contentSize = g[noContentChildren];
            int cookieSize = s[sIndex];
            if(contentSize <= cookieSize){
                noContentChildren++; sIndex++;
            }
            else{
                sIndex++;
            }
        }
        return noContentChildren;
    }
}

File: javascript/455. Assign Cookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function (g, s) {
  g.sort(sortNum);
  s.sort(sortNum);
  var gIndex = 0;
  var sIndex = 0;
  while (gIndex < g.length && sIndex < s.length) {
    var gVal = g[gIndex];
    var sVal = s[sIndex];
    if (gVal <= sVal) {
      gIndex += 1;
      sIndex += 1;
    } else {
      sIndex += 1;
    }
  }
  return gIndex;
};

const sortNum = (a, b) => {
  return a - b;
};
